patches/minecraft/net/minecraft/tileentity/TileEntityBeacon.java.patch

```diff
@@ -9,6 +9,18 @@
                      {
                          this.field_146015_k = false;
                          this.field_174909_f.clear();
+                         break;
+                     }
+ 
++                    float[] color = iblockstate.func_177230_c().getBeaconColor(iblockstate, field_145850_b, blockpos$mutableblockpos); if(color != null){afloat=color;}else{
```

Comment:
I know you're trying to cut down the patch size, but would a new line after the `;` kill you?